src/registry/Registry.ts

```diff
@@ -14,13 +16,140 @@ export class Registry {
+			kusama: {},
+			westend: {},
+			xcAssets: {
+				polkadot: [],
+				kusama: [],
+			},
+		};
+	}
+
+	private initializeCurrentChainIdCache() {
+		const currentChainId = getChainIdBySpecName(this, this.specName);
+		if (!this.cache[this.relayChain][currentChainId]) {
+			this.cache[this.relayChain][currentChainId] = {
+				assetsInfo: {},
+				poolPairsInfo: {},
+				foreignAssetsPalletInstance: null,
+				assetsPalletInstance: null,
+				specName: '',
+				tokens: [],
+				foreignAssetsInfo: {},
+			};
+		}
+	}
+
+	/**
+	 * Getter for the foreignAssetsInfo cache.
+	 *
+	 * @param assetId string
+	 */
+	public cacheLookupForeignAsset(
+		assetId: string
+	): ForeignAssetsData | undefined {
+		const currentChainId = getChainIdBySpecName(this, this.specName);
+		this.initializeCurrentChainIdCache();
+
+		if (
+			this.cache[this.relayChain][currentChainId]['foreignAssetsInfo'][assetId]
+		) {
+			return this.cache[this.relayChain][currentChainId]['foreignAssetsInfo'][
+				assetId
+			] as ForeignAssetsData;
+		}
+
+		return undefined;
+	}
+
+	/**
+	 * Setter for the foreignAssetsInfo cache.
+	 *
+	 * @param assetId string
+	 * @param asset ForeignAssetData
+	 */
+	public setForeignAssetInCache(assetId: string, asset: ForeignAssetsData) {
+		const currentChainId = getChainIdBySpecName(this, this.specName);
+		this.initializeCurrentChainIdCache();
+
+		this.cache[this.relayChain][currentChainId]['foreignAssetsInfo'][assetId] =
+			asset;
+	}
+
+	/**
+	 * Getter for the poolPairsInfo cache.
+	 *
+	 * @param assetId string
+	 */
+	public cacheLookupPoolAsset(
+		assetId: string
+	): { lpToken: string; pairInfo: string } | undefined {
+		const currentChainId = getChainIdBySpecName(this, this.specName);
+		this.initializeCurrentChainIdCache();
+
+		if (this.cache[this.relayChain][currentChainId]['poolPairsInfo'][assetId]) {
+			return this.cache[this.relayChain][currentChainId]['poolPairsInfo'][
+				assetId
+			] as { lpToken: string; pairInfo: string };
+		}
+
+		return undefined;
+	}
+
+	/**
+	 * Setter for the poolPairsInfo cache.
+	 *
+	 * @param assetId string
+	 * @param asset { lpToken: string; pairInfo: string }
+	 */
+	public setLiquidPoolTokenInCache(
+		assetId: string,
+		asset: { lpToken: string; pairInfo: string }
+	) {
+		const currentChainId = getChainIdBySpecName(this, this.specName);
+		this.initializeCurrentChainIdCache();
+
+		this.cache[this.relayChain][currentChainId]['poolPairsInfo'][assetId] =
+			asset;
+	}
+
+	/**
+	 * Getter for the assets cache.
+	 *
+	 * @param assetId string
+	 */
+	public cacheLookupAsset(assetId: string): string | undefined {
+		const currentChainId = getChainIdBySpecName(this, this.specName);
+		this.initializeCurrentChainIdCache();
+
+		if (this.cache[this.relayChain][currentChainId]['assetsInfo'][assetId]) {
+			return this.cache[this.relayChain][currentChainId]['assetsInfo'][assetId];
+		}
+
+		return undefined;
+	}
+
+	/**
+	 * Setter for the assets cache.
+	 *
+	 * @param assetId string
```

Comment:
This is a bit confusing between `assetId`, and `asset`. A better description differentiating between the both would be nice.